extensions/certmanager/generator/cmd/generate/generate.go

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package main
+
+import (
+	"fmt"
+	"github.com/fabric8io/kubernetes-client/generator/pkg/schemagen"
+	certmanager "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1"
+	metav1 "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	certmanageracme	"github.com/jetstack/cert-manager/pkg/apis/acme/v1"
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"reflect"
+)
+
+func main() {
+
+	// the CRD List types for which the model should be generated
+	// no other types need to be defined as they are auto discovered
+	crdLists := map[reflect.Type]schemagen.CrdScope{
+		reflect.TypeOf(certmanager.CertificateList{}): schemagen.Namespaced,
+		reflect.TypeOf(certmanager.CertificateRequestList{}): schemagen.Namespaced,
```

Comment:
how do you know which types to add? gut feeling?

Reply:
I checked their provided CRDs: https://github.com/jetstack/cert-manager/tree/master/deploy/crds and added whatever I could